bool templatebool
float64 templatefloat
string templatestring